.github/actions/build-featurenet-chainspec/action.yml

```diff
@@ -0,0 +1,117 @@
+      with:
+        aws-access-key-id: ${{ inputs.aws-access-key-id }}
+        aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
+        aws-region: us-east-1
+
+    - name: Get aleph-node image tag from testnet
+      if: inputs.base-net == 'testnet'
+      shell: bash
+      run: |
+        COMMIT_ID=$(curl -s -H "Content-Type: application/json" \
+          -d '{"id":1, "jsonrpc":"2.0", "method": "system_version"}' https://rpc.test.azero.dev \
+          | jq -r '.result' | cut -d "-" -f 2 | head -c 7)
+        echo $COMMIT_ID
+        echo "ALEPH_NODE_IMAGE=${{ inputs.ecr-public-registry }}aleph-node:$COMMIT_ID" \
+          >> $GITHUB_ENV
+
+    - name: Get aleph-node image tag from mainnet
+      if: inputs.base-net == 'mainnet'
+      shell: bash
+      run: |
+        SYSTEM_VERSION=$(curl -s -H "Content-Type: application/json" \
+          -d '{"id":1, "jsonrpc":"2.0", "method": "system_version"}' https://rpc.test.azero.dev \
+          | jq -r '.result')
+        SUFFIX="-x86_64-linux-gnu"
+        LONG_COMMIT_ID=${SYSTEM_VERSION/%$SUFFIX}
+        COMMIT_ID=${LONG_COMMIT_ID: -7}
+        echo $COMMIT_ID
+        echo "ALEPH_NODE_IMAGE=${{ inputs.ecr-public-registry }}aleph-node:$COMMIT_ID" \
+          >> $GITHUB_ENV
+
+    - name: Generate data dir
+      shell: bash
+      # yamllint disable rule:line-length
+      run: |
+        # sync all validator's keystores from S3
+        aws s3 cp s3://${{ inputs.featurenet-keys-s3bucket-name }}/data data --recursive
+
+        # rename validator paths
+        declare -A \
+          NAMES=([aleph-node-validator-0]=5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY \
+          [aleph-node-validator-1]=5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty \
+          [aleph-node-validator-2]=5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y \
+          [aleph-node-validator-3]=5DAAnrj7VHTznn2AWBemMuyBwZWs6FNFjdyVXUeYum3PTXFy \
+          [aleph-node-validator-4]=5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw)
+
+        for NAME in "${!NAMES[@]}"; do
+          mv -v data/$NAME data/${NAMES[$NAME]}
+        done
+      # yamllint enable rule:line-length
+
+    - name: Generate chainspec from testnet
+      if: inputs.base-net == 'testnet'
+      shell: bash
+      # yamllint disable rule:line-length
+      run: |
+        docker run -v $(pwd)/data:/data --env RUST_BACKTRACE=1 \
+          --entrypoint "/usr/local/bin/aleph-node" $ALEPH_NODE_IMAGE bootstrap-chain --raw \
+          --base-path /data --chain-id a0fenet \
+          --account-ids 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY,5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty,5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y,5DAAnrj7VHTznn2AWBemMuyBwZWs6FNFjdyVXUeYum3PTXFy,5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw \
+          --sudo-account-id 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY \
+          --faucet-account-id 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY > chainspec.json
```

Comment:
Please consider adding rich accounts as per https://github.com/Cardinal-Cryptography/aleph-node/blob/main/.github/workflows/deploy-to-devnet.yml#L104 , also extract account IDS to separate vars if possible, please :pray: https://github.com/Cardinal-Cryptography/aleph-node/blob/main/.github/workflows/deploy-to-devnet.yml#L94-L97